pkg/loki/modules.go

```diff
@@ -881,7 +881,7 @@ func (t *Loki) initCompactor() (services.Service, error) {
 
 	if t.Cfg.CompactorConfig.RetentionEnabled {
```

Comment:
This could be simplified if we rename the `FilteringEnabled` method to `DeleteEnabled`, remove the switch statement, and changing the condition to `if t.Cfg.CompactorConfig.RetentionEnabled && t.compactor.DeleteE
nabled()`.